src/Core/Utilities/ServiceContainer.cs

```diff
@@ -1,14 +1,15 @@
-﻿using Bit.Core.Abstractions;
-using Bit.Core.Services;
-using System;
-using System.Collections.Generic;
+﻿using System;
+using System.Collections.Concurrent;
+using System.Text;
 using System.Threading.Tasks;
+using Bit.Core.Abstractions;
+using Bit.Core.Services;
 
 namespace Bit.Core.Utilities
 {
     public static class ServiceContainer
     {
-        public static Dictionary<string, object> RegisteredServices { get; set; } = new Dictionary<string, object>();
+        public static ConcurrentDictionary<string, object> RegisteredServices { get; set; } = new ConcurrentDictionary<string, object>();
```

Comment:
Can't this be private?

Reply:
It could be private but currently it's being used from outside to check whether there are services registered. I could change the callers logic to just call a `HasAnyRegistrations()` method/property of the `ServiceContainer`. 
Not sure if we can use the `bool Inited` current property or if we might get into a wall because it's being set to `true` just at the beginning of the `Init` method instead of at the end, thus it's just working to prevent calling Init several times right now and not to check whether all services have been registered, so if we use this property we might get some timing issues on some places, so a different property/method should be better for this that checks directly against the collection or at the end of the registrations.